src/Mod/Arch/archobjects/reinforcement_generic.py

```diff
@@ -0,0 +1,252 @@
+    Should ReinforcementLattice in a separate module?
+
+    Additional Attributes
+    ---------------------
+    BaseRebar : App::PropertyLink
+        the rebar base object
+    RebarPlacements : App::PropertyPlacementList
+        Placement of each rebar of the reinforcement
+    BasePlacement : App::PropertyPlacement
+        on base rebar could be used in many reinforcements, but the rotations
+        might be different in the reinforcements. This placement is applied
+        relative to each placement in RebarPlacements. Thus it is important
+        for rotations only, because each rebar is rotatated instead of the
+        whole reinforcement if applied in Placement attribute. A translation
+        could be applied eitheron BasePlacement or Placement attribute.
+    """
+
+    def __init__(
+        self,
+        obj
+    ):
+        super(ReinforcementGeneric, self).__init__(obj)
+        self.setProperties(obj)
+        obj.IfcType = "Reinforcing Bar"
+
+    def setProperties(
+        self,
+        obj
+    ):
+        self.Type = "ReinforcementGeneric"
+        pl = obj.PropertiesList
+
+        # ArchComponent properties will be inherited
+        # lots of them are not needed, hide them in editor ATM
+        obj.setEditorMode("Additions", 2)
+        obj.setEditorMode("Axis", 2)
+        obj.setEditorMode("Base", 2)
+        obj.setEditorMode("HorizontalArea", 2)
+        obj.setEditorMode("Material", 2)
+        obj.setEditorMode("MoveBase", 2)
+        obj.setEditorMode("MoveWithHost", 2)
+        obj.setEditorMode("PerimeterLength", 2)
+        obj.setEditorMode("StandardCode", 2)  # useful, but not used ATM
+        obj.setEditorMode("Subtractions", 2)
+        obj.setEditorMode("VerticalArea", 2)
+        # CloneOf makes sense if two rebars are identical
+        # but need different MarkNumber
+
+        # New properties
+
+        # BaseRebar
+        if "BaseRebar" not in pl:
+            obj.addProperty(
+                "App::PropertyLink",
+                "BaseRebar",
+                "Reinforcement",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    "Base rebar for this reinforcement"
+                )
+            )
+
+        # RebarPlacements
+        if "RebarPlacements" not in pl:
+            obj.addProperty(
+                "App::PropertyPlacementList",
+                "RebarPlacements",
+                "Reinforcement",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    "Placement for each rebar of the reinforcement"
+                )
+            )
+        # TODO: Why ist this property not shown in PropertyEditor
+
+        # BasePlacement
+        if "BasePlacement" not in pl:
+            obj.addProperty(
+                "App::PropertyPlacement",
+                "BasePlacement",
+                "Reinforcement",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    (
```

Comment:
No parentheses.

Reply:
Aren't they needed because the following string is two lines lone but only one string.

Reply:
No, Python can concatenate strings automatically, as long as they are inside a previous set of parentheses.

Reply:
ahh cool I did not know about. But if they will be removed code formating does not look good anymore. In rare cases I do not aggree with you. Apart from FreeCAD version number the Python parantheses are the only one I know of ATM.

cheers bernd

Reply:
Grrr. You know I dislike your formatting very much, Mr. Bernd. Too many parentheses.